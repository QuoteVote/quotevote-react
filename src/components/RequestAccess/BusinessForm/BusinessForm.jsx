import { makeStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import CardContent from '@material-ui/core/CardContent'
import { TextField } from '@material-ui/core'
import PropTypes from 'prop-types'
import requestAccessStyles from '../requestAccessStyles'
import PaymentMethod from '../PaymentMethod/PaymentMethod'

const useStyles = makeStyles(requestAccessStyles)

const BusinessForm = (props) => {
  const classes = useStyles()
  const {
    requestInviteSuccessful,
    handleSubmit,
    isContinued,
    onContinue,
    errors,
    register,
    setCardDetails,
    cardDetails,
    onSubmit,
    errorMessage,
    loading,
  } = props

  return (
    <Grid container justify="center" style={{ marginRight: 24 }} spacing={2}>
      <Grid item xs={12}>
        <Typography align="center" className={classes.header}>
          {requestInviteSuccessful ? 'Thank you for' : 'Get access to your'}
          {' '}
          <span className={classes.header} style={{ color: '#00cf6e' }}>
            {requestInviteSuccessful ? 'joining us' : 'Business Plan!'}
          </span>
        </Typography>
      </Grid>
      <Grid item xs={12} hidden={requestInviteSuccessful}>
        <Typography align="center" className={classes.subHeader}>
          You are one step away from
          {' '}
          <b>
            unlimited access
          </b>
          {' '}
          to voxPOP
        </Typography>
      </Grid>
      <Grid item xs={12} style={{ marginTop: requestInviteSuccessful ? '4%' : '2%' }}>
        <Grid container spacing={2}>
          <Grid container item xs={12} md={6} justify="center" alignItems="center">
            <img
              alt="Business Illustration"
              height={500}
              src="/assets/Illustration.png"
              style={{
                width: '400px',
                height: '265px',
                objectFit: 'contain',
              }}
            />
          </Grid>
          {requestInviteSuccessful ? (
            <Grid container item xs={12} md={6} justify="center" alignItems="center">
              <Typography className={classes.message}>
                <b>You selected the Business Plan</b>
                , and we
                <br />
                are excited to talk with you.
                <br />
                <br />
                When an account becomes available, an
                <br />
                invite will be sent to the email address you
                <br />
                provided.
              </Typography>
            </Grid>
          ) : (
            <Grid item container xs={12} md={6} spacing={2}>
              <Grid item xs={12}>
                <Card>
                  <CardHeader
                    avatar={(
                      <Typography className={classes.stepNumber}>
                        1
                      </Typography>
                    )}
                    title={(
                      <Typography
                        style={{
                          font: 'Roboto',
                          fontsize: '18px',
                          lineHeight: 1.56,
                        }}
                      >
                        Your Personal Info
                      </Typography>
                    )}
                  />
                  {!isContinued && (
                    <form onSubmit={handleSubmit(onContinue)}>
                      <CardContent>
                        <Grid container spacing={2}>
                          <Grid item xs={6}>
                            <TextField
                              fullWidth
                              required
                              label="Full Name"
                              name="fullName"
                              id="fullName"
                              error={errors.fullName}
                              helperText={errors.fullName && errors.fullName.message}
                              inputRef={register({
                                required: 'Full Name is required',
                                minLength: {
                                  value: 1,
                                  message: 'Full Name should be more than 1 character',
                                },
                                maxLength: {
                                  value: 20,
                                  message: 'Full Name should be less than twenty characters',
                                },
                              })}
                            />
                          </Grid>
                          <Grid item xs={6}>
                            <TextField
                              fullWidth
                              required
                              label="Company Name"
                              name="companyName"
                              id="companyName"
                              error={errors.companyName}
                              helperText={errors.companyName && errors.companyName.message}
                              inputRef={register({
                                required: 'Company Name is required',
                                minLength: {
                                  value: 1,
                                  message: 'Company Name should be more than 1 character',
                                },
                                maxLength: {
                                  value: 20,
                                  message: 'Company Name should be less than twenty characters',
                                },
                              })}
                            />
                          </Grid>
                          <Grid item xs={12}>
                            <TextField
                              fullWidth
                              required
                              label="Email"
                              name="email"
                              id="email"
                              error={errors.email}
                              helperText={errors.email && errors.email.message}
                              inputRef={register({
                                required: 'Email is required',
                                pattern: {
                                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                                  message: 'Invalid email address',
                                },
                              })}
                            />
                          </Grid>
                          <Grid item xs={12}>
                            <Button variant="contained" className={classes.greenBtn} type="submit">
                              Continue
                            </Button>
                          </Grid>
                        </Grid>
                      </CardContent>
                    </form>
                  )}
                </Card>
              </Grid>
              <Grid item xs={12}>
                <PaymentMethod
                  cardDetails={cardDetails}
                  onSubmit={onSubmit}
                  isContinued={isContinued}
                  setCardDetails={setCardDetails}
                  errorMessage={errorMessage}
                  loading={loading}
                />
              </Grid>
            </Grid>
          )}
        </Grid>
      </Grid>
    </Grid>
  )
}

BusinessForm.propTypes = {
  requestInviteSuccessful: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isContinued: PropTypes.bool.isRequired,
  onContinue: PropTypes.func.isRequired,
  errors: PropTypes.any.isRequired,
  register: PropTypes.any.isRequired,
  setCardDetails: PropTypes.func.isRequired,
  cardDetails: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  errorMessage: PropTypes.any,
  loading: PropTypes.bool,
}
export default BusinessForm
