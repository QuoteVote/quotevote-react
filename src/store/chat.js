/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

const chat = createSlice({
  name: 'chat',
  initialState: {
    submitting: false,
    selectedRoom: null,
  },
  reducers: {
    CHAT_SUBMITTING: (state, action) => {
      state.submitting = action.payload
    },
    SELECTED_CHAT_ROOM: (state, action) => {
      state.selectedRoom = action.payload
    },
  },
})

export const { CHAT_SUBMITTING, SELECTED_CHAT_ROOM } = chat.actions

export default chat.reducer
