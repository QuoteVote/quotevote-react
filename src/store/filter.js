/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

const uiSlice = createSlice({
  name: 'filter',
  initialState: {
    filter: {
      visibility: false,
      value: ['POSTED'],
    },
    date: {
      visibility: false,
      value: '',
    },
    search: {
      visibility: false,
      value: '',
    },
  },
  reducers: {
    FILTER_VISIBILITY: (state, action) => {
      state.filter.visibility = action.payload
    },
    FILTER_VALUE: (state, action) => {
      state.filter.value = action.payload
    },
    DATE_VISIBILITY: (state, action) => {
      state.date.visibility = action.payload
    },
    DATE_VALUE: (state, action) => {
      state.date.value = action.payload
    },
    SEARCH_VISIBILITY: (state, action) => {
      state.search.visibility = action.payload
    },
    SEARCH_VALUE: (state, action) => {
      state.search.value = action.payload
    },
  },
})

export const {
  FILTER_VISIBILITY,
  FILTER_VALUE,
  DATE_VISIBILITY,
  DATE_VALUE,
  SEARCH_VISIBILITY,
  SEARCH_VALUE,
} = uiSlice.actions

export default uiSlice.reducer
