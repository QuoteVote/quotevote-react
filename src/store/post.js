/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

const postSlice = createSlice({
  name: 'post',
  initialState: {
    selectedPost: {
      id: null,
    },
  },
  reducers: {
    SET_SELECTED_POST: (state, action) => {
      state.selectedPost.id = action.payload
    },
  },
})

export const { SET_SELECTED_POST } = postSlice.actions

export default postSlice.reducer
