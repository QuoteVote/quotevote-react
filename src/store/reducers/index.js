export { default as postReducer } from './postReducer'
export { default as loginReducer } from './loginReducer'
export { default as chatReducer } from './chatReducer'
export { default as appReducer } from './appReducer'
