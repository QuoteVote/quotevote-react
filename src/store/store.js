import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import postReducer from 'store/post'
import userReducer from 'store/user'

const rootReducer = combineReducers({
  user: userReducer,
  post: postReducer,
})

// Add middleware the end of the array
const middleware = [...getDefaultMiddleware()]

const devMode = process.env.NODE_ENV === 'development'

export default configureStore({
  reducer: rootReducer,
  middleware,
  devTools: devMode,
})
