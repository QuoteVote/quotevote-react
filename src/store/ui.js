/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

export const uiInitialState = {
  filter: {
    visibility: false,
    value: '',
  },
  date: {
    visibility: false,
    value: '',
  },
  search: {
    visibility: false,
    value: '',
  },
  selectedPost: {
    id: null,
  },
  selectedPage: 'home',
  hiddenPosts: [],
  snackbar: {
    open: false,
    type: '',
    message: '',
  },
  selectedPlan: 'personal',
  focusedComment: null,
  sharedComment: null,
}

const uiSlice = createSlice({
  name: 'ui',
  initialState: uiInitialState,
  reducers: {
    SET_SELECTED_POST: (state, action) => {
      state.selectedPost.id = action.payload
    },
    SET_SELECTED_PAGE: (state, action) => {
      state.selectedPage = action.payload
    },
    SET_HIDDEN_POSTS: (state, action) => {
      state.hiddenPosts = [...state.hiddenPosts, action.payload]
    },
    SET_SNACKBAR: (state, action) => {
      state.snackbar = action.payload
    },
    SET_SELECTED_PLAN: (state, action) => {
      state.selectedPlan = action.payload
    },
    SET_FOCUSED_COMMENT: (state, action) => {
      state.focusedComment = action.payload
    },
    SET_SHARED_COMMENT: (state, action) => {
      state.sharedComment = action.payload
    },
  },
})

export const {
  SET_SELECTED_POST,
  SET_SELECTED_PAGE,
  SET_HIDDEN_POSTS,
  SET_SNACKBAR,
  SET_SELECTED_PLAN,
  SET_FOCUSED_COMMENT,
  SET_SHARED_COMMENT,
} = uiSlice.actions

export default uiSlice.reducer
