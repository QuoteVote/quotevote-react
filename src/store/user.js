/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'
import axios from 'axios'
import jwt from 'jsonwebtoken'

const userSlice = createSlice({
  name: 'user',
  initialState: {
    loading: false,
    loginError: null,
    data: {},
  },
  reducers: {
    USER_LOGIN_REQUEST: (state) => {
      state.loading = true
    },
    USER_LOGIN_SUCCESS: (state, action) => {
      const { data, loading, loginError } = action.payload
      state.data = data
      state.loading = loading
      state.loginError = loginError
    },
    USER_LOGIN_FAILURE: (state, action) => {
      const { loginError, loading } = action.payload
      state.loading = loading
      state.loginError = loginError
    },
    USER_LOGOUT: (state) => {
      state.loading = false
      state.isLoggedIn = false
      state.data = {}
    },
    USER_TOKEN_VALIDATION: (state) => {
      state.loading = true
    },
    USER_TOKEN_VALIDATED: (state) => {
      state.loading = false
    },
    USER_UPDATE_AVATAR: (state, action) => {
      state.data.avatar = action.payload
    },
    SET_USER_DATA: (state, action) => {
      state.data = action.payload
    },
  },
})

// Do not export the actions.
// Create functions, then export them
const { actions } = userSlice

// Private function(s) for actions
const getToken = async (username, password) => {
  try {
    return await axios.post(
      `${process.env.REACT_APP_SERVER}/login`,
      {
        username,
        password,
      },
      {
        headers: {
          'Content-Type': 'application/json',
        },
      }
    )
  } catch (err) {
    return { error: err }
  }
}

// Public functions that dispatch multiple actions
export const userLogin = async (username, password, dispatch, history) => {
  dispatch(actions.USER_LOGIN_REQUEST())
  dispatch(
    actions.USER_LOGIN_FAILURE({ loginError: null, loading: true })
  )
  const result = await getToken(username, password)

  if ('error' in result) {
    const serverConnectionRefuseError = {
      data: { message: 'Connection refuse' },
    }
    const errorMessage =
      typeof result.error.response !== 'undefined' ?
        result.error.response.data.message :
        serverConnectionRefuseError
    dispatch(
      actions.USER_LOGIN_FAILURE({ loginError: errorMessage, loading: false })
    )
  } else {
    const { token, user } = result.data
    localStorage.setItem('token', token)
    dispatch(
      actions.USER_LOGIN_SUCCESS({
        data: user,
        loading: false,
        loginError: null,
      })
    )

    history.push('/hhsb/Profile/hhsb/avatar')
  }
}

export function tokenValidator(dispatch) {
  dispatch(actions.USER_TOKEN_VALIDATION())
  const storedToken = localStorage.getItem('token')
  return jwt.verify(storedToken, 'HHSB', (err) => {
    if (err) {
      localStorage.removeItem('token')
      dispatch(actions.USER_LOGOUT())
      // console.log("Error", err);
      return false
    }
    // console.log(decoded);
    dispatch(actions.USER_TOKEN_VALIDATED())
    return true
  })
}

export function clearToken(dispatch) {
  dispatch(actions.USER_LOGOUT())
  localStorage.removeItem('token')
}

export function updateAvatar(dispatch, avatar) {
  dispatch(actions.USER_UPDATE_AVATAR(avatar))
}

export const {
  USER_LOGIN_FAILURE,
  USER_LOGIN_REQUEST,
  USER_LOGIN_SUCCESS,
  SET_USER_DATA,
  USER_TOKEN_VALIDATED,
  USER_TOKEN_VALIDATION,
  USER_UPDATE_AVATAR,
} = userSlice.actions

export default userSlice.reducer
