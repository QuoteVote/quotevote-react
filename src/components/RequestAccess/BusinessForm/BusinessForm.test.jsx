import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import BusinessForm from './BusinessForm'
import store from '../../../store/store'

const props = {
  handleSubmit: jest.fn(),
  register: jest.fn(),
  errors: {
    fullName: '',
  },
}

const BusinessFormWrapper = () => (
  <Provider store={store}>
    <BusinessForm {...props} />
  </Provider>
)

describe('BusinessForm test -', () => {
  it('renders correctly', () => {
    const { container } = render(<BusinessFormWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
