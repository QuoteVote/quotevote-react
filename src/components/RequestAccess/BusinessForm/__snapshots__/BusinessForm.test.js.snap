// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessForm test - renders correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  style="margin-right: 24px;"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <p
      class="MuiTypography-root makeStyles-header-2 MuiTypography-body1 MuiTypography-alignCenter"
    >
      Get access to your
       
      <span
        class="makeStyles-header-2"
        style="color: rgb(0, 207, 110);"
      >
        Business Plan!
      </span>
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <p
      class="MuiTypography-root makeStyles-subHeader-3 MuiTypography-body1 MuiTypography-alignCenter"
    >
      You are one step away from
       
      <b>
        unlimited access
      </b>
       
      to voxPOP
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    style="margin-top: 2%;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
      >
        <img
          alt="Illustration.png"
          height="500"
          src="Illustration.png"
          style="width: 400px; height: 265px; object-fit: contain;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-avatar"
              >
                <p
                  class="MuiTypography-root makeStyles-stepNumber-4 MuiTypography-body1"
                >
                  1
                </p>
              </div>
              <div
                class="MuiCardHeader-content"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                  style="font: Roboto; line-height: 1.56;"
                >
                  Your Personal Info
                </p>
              </div>
            </div>
            <form>
              <div
                class="MuiCardContent-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                        data-shrink="false"
                        for="fullName"
                        id="fullName-label"
                      >
                        Full Name
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid=""
                          class="MuiInputBase-input MuiInput-input"
                          id="fullName"
                          name="fullName"
                          required=""
                          type="text"
                          value=""
                        />
                      </div>
                      
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                        data-shrink="false"
                        for="companyName"
                        id="companyName-label"
                      >
                        Company Name
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="companyName"
                          name="companyName"
                          required=""
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                        data-shrink="false"
                        for="email"
                        id="email-label"
                      >
                        Email
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="email"
                          name="email"
                          required=""
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-greenBtn-1"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Continue
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <form>
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <p
                    class="MuiTypography-root makeStyles-stepNumber-22 MuiTypography-body1"
                  >
                    2
                  </p>
                </div>
                <div
                  class="MuiCardHeader-content"
                >
                  <p
                    class="MuiTypography-root makeStyles-stepName-23 MuiTypography-body1"
                  >
                    Would you like to support this app?
                  </p>
                </div>
              </div>
              <input
                name="cardNumber"
                type="hidden"
              />
              <input
                name="expiry"
                type="hidden"
              />
              <input
                name="cvv"
                type="hidden"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
